boolean_parser/mixins/sqla.py

```diff
@@ -166,7 +194,35 @@ def _filter_one(self, model, field=None, condition=None):
         elif self.operator == '>=':
             condition = lower_field.__ge__(lower_value)
         elif self.operator == '!=':
-            condition = lower_field.__ne__(lower_value)
+            if isinstance(field.type, sqltypes.TEXT) or \
+                isinstance(field.type, sqltypes.VARCHAR) or \
+                isinstance(field.type, sqltypes.String):
+                field = getattr(model, self.name)
+                value = self.value
+                if value.lower() == 'null':
+                    condition = field.isnot(None)
+                else:
+                    condition = lower_field.__ne__(lower_value)
+            elif isinstance(field.type, sqltypes.BOOLEAN) or \
+                isinstance(field.type, sqltypes.Boolean):
+                field = getattr(model, self.name)
+                value = self.value
+                if value.lower() == 'null':
+                    condition = field.isnot(None)
+                else:
+                    condition = lower_field.__ne__(lower_value)
+            elif isinstance(field.type, sqltypes.DATE) or \
+                isinstance(field.type, sqltypes.DATETIME) or \
+                isinstance(field.type, sqltypes.Date) or \
+                isinstance(field.type, sqltypes.DateTime):
+                field = getattr(model, self.name)
+                value = self.value
+                if value.lower() == 'null':
+                    condition = field.isnot(None)
+                else:
+                    condition = lower_field.__ne__(lower_value)
+            else:
+                condition = lower_field.__ne__(lower_value)
```

Comment:
This bit of code looks the same as how it's used for the `==` operator at lines 159-187, with the exception of the `field` function.  I'm always in favor of not duplicating code for easier maintainability and readability.  Maybe we could move this code into a new common method that accepts the relevant inputs?

Reply:
My lack of understanding of the code led to quite a lot of duplication. It should be much more straightforward now.